### PR TITLE
Added possibility of narrowing users by working hours in Automatic As…

### PR DESCRIPTION
**Pull request by konadrian**
Added possibility of narrowing users by working hours in Automatic Assigned

### CODE REVIEW

**Comment by ghost** [comment]
## Black Duck Security Report

Merging #17201 into developer will not change security risk.

[Click here to see full report](https://copilot.blackducksoftware.com/github/repos/YetiForceCompany/YetiForceCRM/pull-requests/17201)

File: app/AutoAssign.php [hunk 1]
```diff
@@ -346,6 +346,11 @@ public function getAvailableUsersQuery(): QueryGenerator
 			->addCondition('status', 'Active', 'e')
 			->addCondition('available', 1, 'e')
 			->addCondition('auto_assign', 1, 'e');
+		if ($this->get('working_hours')) {
+			$currentTime = date('H:i');
+			$queryGenerator->addNativeCondition(['<=', 'start_hour', $currentTime])
+				->addNativeCondition(['>', 'end_hour', $currentTime]);
```
Comment (rskrzypczak) [inline app/AutoAssign.php]: why u use addNativeCondition instead of addCondition?

**Review by rskrzypczak — CHANGES_REQUESTED** [review]

File: app/AutoAssign.php [hunk 2]
```diff
@@ -346,6 +346,11 @@ public function getAvailableUsersQuery(): QueryGenerator
 			->addCondition('status', 'Active', 'e')
 			->addCondition('available', 1, 'e')
 			->addCondition('auto_assign', 1, 'e');
+		if ($this->get('working_hours')) {
+			$currentTime = date('H:i');
+			$queryGenerator->addNativeCondition(['<=', 'start_hour', $currentTime])
```
Comment (rskrzypczak) [inline app/AutoAssign.php]:
```suggestion
			$queryGenerator->addNativeCondition(['<=', $queryGenerator->getColumnName('start_hour'), $currentTime])
```

**Review by rskrzypczak — CHANGES_REQUESTED** [review]

File: app/AutoAssign.php (hunk 1 above)
Comment (konadrian) [reply app/AutoAssign.php]: Start and end hours are a picklist fields, not a time

**Pull request merged**